src/v/model/record_batch_reader.h

```diff
@@ -280,6 +280,36 @@ make_memory_record_batch_reader(model::record_batch b) {
     return make_memory_record_batch_reader(std::move(batches));
 }
 
+// clang-format off
+template<typename Func>
+CONCEPT(requires requires(Func f, model::record_batch&& batch) {
+    { f(batch) } -> std::same_as<model::record_batch>;
+})
+// clang-format on
+ss::future<record_batch_reader::data_t> transform_reader_to_memory(
```

Comment:
@graphcareful did you already have something like this in the coproc tree?

Reply:
I wrote sort of an inverse. The one I wrote takes as input a record_batch_reader::data_t and returns a record_batch_reader.